iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/BatteryAdderImpl.java

```diff
@@ -89,20 +90,21 @@ public BatteryAdderImpl setMaxP(double maxP) {
      */
     @Override
     public BatteryImpl add() {
+        NetworkImpl network = getNetwork();
         String id = checkAndGetUniqueId();
         TerminalExt terminal = checkAndGetTerminal();
-        ValidationUtil.checkP0(this, p0);
-        ValidationUtil.checkQ0(this, q0);
+        network.setValidationLevelIfGreaterThan(ValidationUtil.checkP0(this, p0, network.getMinValidationLevel().compareTo(ValidationLevel.LOADFLOW) >= 0));
```

Comment:
Maybe we could change the ValidationUtil API to avoid this pattern `network.getMinValidationLevel().compareTo(ValidationLevel.LOADFLOW) >= 0` everywhere in the code?